primap2/_aggregate.py

```diff
@@ -220,6 +220,99 @@ def fill_all_na(self, dim: Iterable[Hashable] | str, value=0) -> xr.DataArray:
+        agg_info:
+            dict of the following form
+            {
+                <coord1>: {
+                    <new_value> : {
+                        'sources': [source_values],
+                        <add_coord_name>: <value for additional coordinate> (optional),
+                        'tolerance': <non-default tolerance> (optional),
+                        'filter': <filter in pr.loc style> (optional),
+                    },
+                },
+                <coord2>: {...},
+                ...
+            }
+
+        tolerance:
+            non-default tolerance for merging (default = 0.01 (1%))
+
+        Returns
+        -------
+            xr.DataArray with aggregated values for coordinates / dimensions as
+            specified in the agg_info dict
+
+        """
+
+        # dequantify for improved speed. unit handling is not necessary as we
+        # work with one DataArray and thus the unit is the same for all
+        # timeseries that are aggregated
+        da_out = self._da.pr.dequantify()
+
+        for coordinate in agg_info.keys():
+            aggregation_rules = agg_info[coordinate]
+            full_coord_name = da_out.pr.dim_alias_translations[coordinate]
+            for value_to_aggregate in aggregation_rules.keys():
+                rule = aggregation_rules[value_to_aggregate].copy()
+                if "tolerance" in rule.keys():
+                    rule_tolerance = rule.pop("tolerance")
+                else:
+                    rule_tolerance = tolerance
+                if "filter" in rule.keys():
+                    filter = rule.pop("filter")
+                else:
+                    filter = {}
+
+                source_values = rule.pop("sources")
+
+                filter.update({coordinate: source_values})
+                data_agg = da_out.pr.loc[filter].pr.sum(
+                    dim=coordinate, skipna=True, min_count=1
+                )
+                if not data_agg.isnull().all().data:
+                    data_agg = data_agg.expand_dims([full_coord_name])
+                    data_agg = data_agg.assign_coords(
+                        coords={
+                            full_coord_name: (full_coord_name, [value_to_aggregate])
+                        }
+                    )
+                    for add_coord in rule.keys():
+                        if add_coord in data_agg.coords:
+                            add_coord_value = rule[add_coord]
+                            data_agg = data_agg.assign_coords(
+                                coords={add_coord: (full_coord_name, [add_coord_value])}
+                            )
+                        else:
+                            raise ValueError(
+                                f"Additional coordinate {add_coord} specified but not "
+                                f"present in data"
+                            )
+
+                    da_out = da_out.pr.merge(data_agg, tolerance=rule_tolerance)
```

Comment:
I'm wondering if it makes sense to add a `merge`-like mode to `pr.set`. In theory, the category aggregation sounds like something which should be a `pr.sum` and a `pr.set` and nothing else, but here you have to do quite a lot of set-up to turn `data_agg` into something which can be consumed by `pr.merge`.

Reply:
Yes, that would make things easier. But I think for this case I could also just use my `ds.pr.add_dim` function instead of doing the same steps manually